orangecontrib/timeseries/timeseries.py

```diff
@@ -19,16 +19,117 @@ def __init__(self, *args, **kwargs):
+                                       if var.is_continuous)
+            return cls.make_timeseries_from_continuous_var(table, continuous_variable)
+        except (StopIteration, AttributeError):
+            pass
+        # Fallback to sequential
+        return cls.make_timeseries_from_sequence(table)
+
     @classmethod
-    def from_data_table(cls, table):
-        return table if isinstance(table, Timeseries) else Timeseries(table)
+    def from_domain(cls, *args, detect_time_variable=False, **kwargs):
+        table = Table.from_domain(*args, **kwargs)
+        return cls.from_data_table(table, detect_time_variable=detect_time_variable)
+
+    @classmethod
+    def from_table(cls, domain, source, *args, detect_time_variable=False, **kwargs):
+        if not isinstance(source, Timeseries):
+            table = Table.from_table(domain, source, *args, **kwargs)
+            return cls.from_data_table(table, detect_time_variable=detect_time_variable)
+        return super().from_table(domain, source, *args, **kwargs)
+
+    @classmethod
+    def from_numpy(cls, *args, detect_time_variable=False, **kwargs):
+        table = Table.from_numpy(*args, **kwargs)
+        return cls.from_data_table(table, detect_time_variable=detect_time_variable)
+
+    @classmethod
+    def from_list(cls, *args, detect_time_variable=False, **kwargs):
+        table = Table.from_list(*args, **kwargs)
+        return cls.from_data_table(table, detect_time_variable=detect_time_variable)
+
+    @classmethod
+    def from_file(cls, *args, detect_time_variable=False, **kwargs):
+        table = Table.from_file(*args, **kwargs)
+        return cls.from_data_table(table, detect_time_variable=detect_time_variable)
+
+    @classmethod
+    def from_url(cls, *args, detect_time_variable=False, **kwargs):
+        table = Table.from_url(*args, **kwargs)
+        return cls.from_data_table(table, detect_time_variable=detect_time_variable)
+
+    @classmethod
+    def make_timeseries_from_sequence(cls, table):
+        attrs = table.domain.attributes
+        cvars = table.domain.class_vars
+        metas = table.domain.metas
+        X = table.X
+        Y = np.column_stack((table.Y,))  # make 2d
+        M = table.metas
+
+        # Uniqueify seq name
+        for i in itertools.chain(('',), range(10)):
+            name = '__seq__' + str(i)
+            if name not in table.domain:
+                break
+        # Create new time variable, values 1 to len(data + 1)
+        time_var = ContinuousVariable(name)
+        attrs = attrs.__class__((time_var,)) + attrs
+        X = np.column_stack((np.arange(1, len(table) + 1), X))
+        table = Table(Domain(attrs, cvars, metas), X, Y, M)
+
+        ts = super(Timeseries, cls).from_table(table.domain, table)
+        ts.time_variable = time_var
+        return ts
+
+    @classmethod
+    def make_timeseries_from_continuous_var(cls, table, attr_name):
+        # Or make a sequence attribute from one of the existing attributes,
```

Comment:
Or make a sequence...? Or as opposed to what? 😄 Probably the comment should be a standalone?